Dockerfile

```diff
@@ -24,33 +27,48 @@ LABEL maintainer="Microsoft" \
 
 WORKDIR azure-cli
 COPY . /azure-cli
-# pip wheel - required for CLI packaging
-# jmespath-terminal - we include jpterm as a useful tool
-RUN pip install --no-cache-dir --upgrade pip wheel jmespath-terminal
+
 # bash gcc make openssl-dev libffi-dev musl-dev - dependencies required for CLI
-# jq - we include jq as a useful tool
 # openssh - included for ssh-keygen
 # ca-certificates
 # wget - required for installing jp
-RUN apk add --no-cache bash gcc make openssl-dev libffi-dev musl-dev jq openssh \
-    ca-certificates wget openssl git && update-ca-certificates
-# We also, install jp
-RUN wget https://github.com/jmespath/jp/releases/download/0.1.2/jp-linux-amd64 -qO /usr/local/bin/jp && chmod +x /usr/local/bin/jp
-
+# jq - we include jq as a useful tool
+RUN apk add --no-cache bash openssh ca-certificates jq wget openssl git \
+ && apk add --no-cache --virtual .build-deps gcc make openssl-dev libffi-dev musl-dev \
+ && update-ca-certificates \
+ && wget https://github.com/jmespath/jp/releases/download/${JP_LATEST_VERSION}/jp-linux-amd64 -qO /usr/local/bin/jp \
+ && chmod +x /usr/local/bin/jp \
+# pip wheel - required for CLI packaging
+# jmespath-terminal - we include jpterm as a useful tool
+ && pip install --no-cache-dir --upgrade jmespath-terminal -r requirements.txt \
 # 1. Build packages and store in tmp dir
 # 2. Install the cli and the other command modules that weren't included
 # 3. Temporary fix - install azure-nspkg to remove import of pkg_resources in azure/__init__.py (to improve performance)
-RUN /bin/bash -c 'TMP_PKG_DIR=$(mktemp -d); \
+&& /bin/bash -c 'TMP_PKG_DIR=$(mktemp -d); \
     for d in src/azure-cli src/azure-cli-core src/azure-cli-nspkg src/azure-cli-command_modules-nspkg src/command_modules/azure-cli-*/; \
     do cd $d; echo $d; python setup.py bdist_wheel -d $TMP_PKG_DIR; cd -; \
     done; \
     [ -d privates ] && cp privates/*.whl $TMP_PKG_DIR; \
     all_modules=`find $TMP_PKG_DIR -name "*.whl"`; \
     pip install --no-cache-dir $all_modules; \
-    pip install --no-cache-dir --force-reinstall --upgrade azure-nspkg azure-mgmt-nspkg;'
-
+    pip install --no-cache-dir --force-reinstall --upgrade azure-nspkg azure-mgmt-nspkg;' \
 # Tab completion
-RUN cat /azure-cli/az.completion > ~/.bashrc
+&& cat /azure-cli/az.completion > ~/.bashrc \
+    && find /usr/local \
+        \( -type d -a -name test -o -name tests \) \
+        -o \( -type f -a -name '*.pyc' -o -name '*.pyo' \) \
+        -exec rm -rf '{}' + \
```

Comment:
Why remove the pyc files?
They'll likely be created again on first run and may slow down first run performance.